content/news/20.4networking-rework.md

```diff
@@ -0,0 +1,224 @@
+If the handler is invoked on the server side, then this is the level the player is in.
+If the handler is invoked on the client side, then this is the client's current level.
+
+##### TaskCompletedHandler
+This is a special contextual value only available to payloads for the configuration, which are specifically registered as such, to indicate that a specific configuration task has been completed, and that the next one can be started.
+
+##### Future additions to the contexts
+We are fully aware that these entry points might not be all information you as a modder need to process a packet. In general, it is pretty easy to extend the interfaces and records. **They were specifically designed to allow for simple PRs in the future to add to them**, so please do not hesitate to create a quick PR to add your needed data to the context.
+
+### Packet Sending
+In tandem with the refactoring of the channel registration mechanic we added new tools and systems to allow you to easier send a custom payload to different targets.
+
+#### PacketDistributor
+This wrapper class now has the ability to process custom payloads solely. Its instances and targets can be passed around since they are immutable. Several methods on extension classes will accept these, to facilitate easy transfer of payloads.
+
+#### Extension objects
+We extended several vanilla types allowing for them to accept payloads as well, not just packets. Examples are chunk sections, listeners, entities and players.
+
+### Netty information
+In contrast with the past, we now store a lot of information related to the connection, for example the negotiated payload types, on the connection object itself. To this accord we added several attributes to store this information in. Although the attributes are considered part of the internal API, if somebody wants to dig through the internal guts, they should feel free to do so, however there is no guarantee that we would change them around in future releases, even outside the BC window.
+
+## Configuration tasks on client join
+### Tasks
+Vanilla now provides a centralized way to perform tasks and jobs that need to be performed when a player joins. The player won't be instantiated or added to a world until these tasks are completed.
+
+Under normal circumstances these tasks are implementations of the `ConfigurationTask` interface, with a single method: `start(Consumer<Packet<?>>)`. However, this is subpar in our situation. Modders should never really have to touch raw Packets to perform their duties, only payloads. And as such it was decided to have modders implement the `ICustomConfigurationTask` interface from NeoForge itself. This provides a wrapper around the `ConfigurationTask` signature and allows for sending payloads instead of packets, by implementing `run(Consumer<CustomPacketPayload>)` instead. The given consumer will then automatically convert the payload to a packet and send it to the client that is being configured.
```

Comment:
```suggestion
Under normal circumstances these tasks are implementations of the `ConfigurationTask` interface, with a single method: `start(Consumer<Packet<?>>)`. However, this is subpar in our situation. Modders should never really have to touch raw Packets to perform their duties, only payloads. And as such it was decided to have modders implement the `ICustomConfigurationTask` interface from NeoForge. This provides a wrapper around the `ConfigurationTask` signature and allows sending payloads instead of packets, by implementing `run(Consumer<CustomPacketPayload>)` instead. The given consumer will then automatically convert the payload to a packet and send it to the client that is being configured.
```